app/models/project.rb

```diff
@@ -73,9 +73,10 @@ def tag_list
   end
 
   def tag_list=(names)
-    self.tags = names.split(",").map do |n|
+    self.tags = names.split(/\s*,\s*/).compact.reject(&:empty?).uniq.map do |n|
```

Comment:
Instead of using `compact.reject(&:empty?)` is it possible to use `.select(&:present?)` here?

Reply:
I think we can. Should we?

Reply:
What was the reason for this change ?

Reply:
i felt `.compact.reject(&:empty?)` seemed a little repetitive for the task that it was trying to accomplish, so suggested doing `.select(&:present?)`. But it's not necessary, we can keep it the way it is.

Reply:
Not that, I meant splitting using regex

Reply:
`names.split(",").map(&:strip)` is one way to go about it.

Reply:
> `names.split(",").map(&:strip)` is one way to go about it and `names.split(/\s*,\s*/)` also does the same thing. I think it is to remove the white space characters around `,`

@tachyons yeah this is the reason, earlier I was just stripping whitespace after I split across `,` but now ` tag` and `tag ` will be different if I split it that way.

Reply:
If `names.split(",").map(&:strip)` this can be used, this is preferable as it is definitely more readable.

Reply:
@shubhankarsharma00  Can we change this to `names.split(",").map(&:strip)` :sweat_smile: ?

Reply:
@ankitkataria, thanks for the suggestion this is definitely more readable.